solr/test-framework/src/java/org/apache/solr/util/EmbeddedSolrServerTestRule.java

```diff
@@ -0,0 +1,199 @@
+            UpdateShardHandlerConfig.DEFAULT_MAXRECOVERYTHREADS);
+
+    return new NodeConfig.NodeConfigBuilder("testNode", solrHome)
+        .setUpdateShardHandlerConfig(updateShardHandlerConfig)
+        .setCoreRootDirectory(LuceneTestCase.createTempDir("cores").toString());
+  }
+
+  public NewCollectionBuilder newCollection(String name) {
+    return new NewCollectionBuilder(name);
+  }
+
+  public class NewCollectionBuilder {
+    private String name;
+    private String configSet;
+    private String configFile;
+    private String schemaFile;
+
+    public NewCollectionBuilder(String name) {
+      this.name = name;
+    }
+
+    public NewCollectionBuilder withConfigSet(String configSet) {
+      // Chop off "/conf" if found -- configSet can be a path.
+      // This is a hack so that we can continue to use ExternalPaths.DEFAULT_CONFIGSET etc. as-is.
+      // TODO FileSystemConfigSetService.locateInstanceDir should have this logic.
+      // Without this, managed resources might be written to
+      // conf/conf/_schema_analysis_stopwords_english.json because SolrResourceLoader points to the
+      // wrong dir.
+      if (configSet != null && configSet.endsWith("/conf")) {
+        configSet = configSet.substring(0, configSet.length() - "/conf".length());
+      }
+
+      this.configSet = configSet;
+      return this;
+    }
+
+    public NewCollectionBuilder withConfigFile(String configFile) {
+      this.configFile = configFile;
+      return this;
+    }
+
+    public NewCollectionBuilder withSchemaFile(String schemaFile) {
+      this.schemaFile = schemaFile;
+      return this;
+    }
+
+    public String getName() {
+      return name;
+    }
+
+    public String getConfigSet() {
+      return configSet;
+    }
+
+    public String getConfigFile() {
+      return configFile;
+    }
+
+    public String getSchemaFile() {
+      return schemaFile;
+    }
+
+    public void create() throws SolrServerException, IOException {
+      EmbeddedSolrServerTestRule.this.create(this);
+    }
+  }
+
+  private void create(NewCollectionBuilder b) throws SolrServerException, IOException {
+
+    client = new EmbeddedSolrServer(container, b.getName());
+
+    CoreAdminRequest.Create req = new CoreAdminRequest.Create();
+    req.setCoreName(b.getName());
+    req.setInstanceDir(b.getName());
+
+    if (b.getConfigSet() != null) {
+      req.setConfigSet(b.getConfigSet());
+    }
+    if (b.getConfigFile() != null) {
+      req.setConfigName(b.getConfigFile());
+    }
+    if (b.getSchemaFile() != null) {
+      req.setSchemaName(b.getSchemaFile());
+    }
+
+    req.process(client);
+  }
+
+  @Override
+  protected void after() {
+    if (container != null) container.shutdown();
+
+    if (clearCoreDirSysProp) {
+      System.clearProperty(CORE_DIR_PROP);
+    }
+  }
+
+  @Override
+  public EmbeddedSolrServer getSolrClient() {
+    if (client == null) {
+      return adminClient;
+    } else {
+      return client;
+    }
```

Comment:
Let's not assume the last call to create a collection is the default client (the `client` field).  Instead, let's establish that `getSolrClient` will always return a client to "collection1".  It should work even if there is such a core loaded by Solr in startSolr (i.e. no call to explicitly create a collection).  Don't return adminClient here either.